ow_plexil/src/plexil-adapter/OwInterface.cpp

```diff
@@ -804,6 +859,44 @@ void OwInterface::guardedMoveAction (double x, double y, double z,
     guarded_move_done_cb<GuardedMove, ow_lander::GuardedMoveResultConstPtr>);
 }
 
+std::vector<double> OwInterface::identifySampleLocation (int num_images, std::string filter_type, int id)
+{ 
+  m_sample_point.clear();
+  got_sample_location_result = false;
+  if (! mark_operation_running (Op_IdentifySampleLocation, id)) return m_sample_point;
+  thread action_thread (&OwInterface::identifySampleLocationAction, this, num_images, filter_type, id);
+  action_thread.detach();
+
+  ros::Rate rate(10);
+  int timeout = 0;
+  /* We wait for a result from the action server or the 5 sec timeout before returning
+  I dont think a timeout is strictly necessary here but because this is blocking
+  I put it in for safety */
```

Comment:
This is our first Plexil command that returns a value, so there is no precedent for how to implement it.  Since the returned value is implemented as that of this function, this function must wait for it.  And since the only Plexil calls of this command are Synchronous, this is okay.   Ideally, the C++ implementation should not block, and thus support asynchronous value return in Plexil (this is allowed there...the assigned variable is UNKNOWN until the value arrives).  But in practice, most value-returning commands would be called synchronously.

In short, the wait/timeout here seems okay, and I don't have anything better to suggest.